src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.service.FileReaderService;
+import core.basesyntax.service.FileWriterService;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.impl.DataConverterImpl;
+import core.basesyntax.service.impl.FileReaderServiceImpl;
+import core.basesyntax.service.impl.FileWriterServiceImpl;
+import core.basesyntax.service.impl.ReportGeneratorImpl;
+import core.basesyntax.service.impl.ShopServiceImpl;
+import core.basesyntax.service.impl.operation.BalanceOperation;
+import core.basesyntax.service.impl.operation.PurchaseOperation;
+import core.basesyntax.service.impl.operation.ReturnOperation;
+import core.basesyntax.service.impl.operation.SupplyOperation;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    private static final String PATH_TO_REPORT_FILES = "src/main/resources/";
+    private static final String DATA_FROM_FILE_NAME = "reportToRead.csv";
+    private static final String DATA_TO_FILE_NAME = "finalReport.csv";
+    private static final FruitDao fruitDao = new FruitDaoImpl();
+
+    public static void main(String[] args) {
```

Comment:
The main method should be responsible for orchestrating the flow of the application by calling services that are independent of each other. However, it seems that the services are tightly coupled and not all are called independently. For instance, the FileReaderService should not be directly aware of the data conversion logic or the specific operations for each transaction.

Reply:
++